### PR TITLE
mail/emailrelay: rework init script and introduce config

### PR DESCRIPTION
**Pull request by p-wassi**
Maintainer: @fededim 
Compile tested: ar71xx
Run tested: ar71xx

Description:
The first commit is just doing some cosmetics, nothing functional, therefore also the PKG_RELEASE
is kept the same:
- Trailing spaces and tabs
- Add newline to end of file
- Add License info to Makefile
- Move maintainer definition to `PKG_MAINTAINER`

The second commit contains deeper changes:
- The given spool directory `/tmp/spool/emailrelay` was created by the install script
  Since this path is on a `tmpfs`, it will be gone at next reboot after installation.
  Therefore dynamically create the default directory `/var/spool/emailrelay` from the init script.
- Switch to a `procd`-style init script
- Currently one has to configure the package by fiddling with the command line inside the
  init script, which seems quite suboptimal. Introduce a real config file in `/etc/config` instead.
  The config currently supports multiple instances of these types:
  - `mode=server`: accept incoming SMTP connection and simply store mails to disk (spool-dir)
  - `mode=proxy`: forward incoming SMTP mails to a smarthost
  - `mode=cmdline`: specify the executable's cmdline on someone's own

 All supported options are given as an example in the installed config file. Check that file for
 reference. Since `emailrelay` has so many different options, one can always use the `extra_cmdline`
 option to specify own parameters that will be passed to the binary additionally.

Further, I'd like to discuss, whether the other binaries `emailrelay-passwd`, `emailrelay-filter-copy`,
`emailrelay-submit` and `emailrelay-poke` should be copied to the target or need to be built at
all. The main functionality is included in the `emailrelay` binary. I don't think that the others would
be used at a regular basis on the target (if ever called at all). I mean is there a real-world scenario,
that is not some special case, where one would need those?

### CODE REVIEW

File: mail/emailrelay/files/emailrelay.config [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#	option server_tls '/etc/path/to/certificate'
+#	option server_auth '/etc/emailrelay.auth'
+#	option client_tls '1'
+#	option client_auth '/etc/emailrelay.auth'
+#	option extra_cmdline ''
+
+config emailrelay 'cmdline'
+	option enabled '0'
+	option mode 'cmdline'
+	# specify all arguments that should be passed to emailrelay here
```
Comment (fededim) [inline mail/emailrelay/files/emailrelay.config]: Please just add a comment with the url for cmdline reference, e.g.
#see http://emailrelay.sourceforge.net/reference.html for command line reference

Comment (fededim) [inline mail/emailrelay/files/emailrelay.config]: Please just add a comment with the url for cmdline reference, e.g.
#see http://emailrelay.sourceforge.net/reference.html for command line reference

**Comment by p-wassi** [comment]
Thanks for reviewing @fededim 
I just added the comment to the config file.

**Comment by fededim** [comment]
Changes are fine for me

**Pull request merged**